src/github_utils.py

```diff
@@ -0,0 +1,132 @@
+    return response.text
+
+
+def get_valid_lines(diff_content):
+    """Parse diff to find valid line numbers with proper initialization"""
+    valid_lines = {}
+    current_file = None
+    line_number = None
+    
+    for line in diff_content.split('\n'):
+        # Reset tracking for new files
+        if line.startswith('diff --git'):
+            current_file = line.split(' b/')[-1].strip()
+            valid_lines[current_file] = set()
+            line_number = None  # Reset for new file
+            continue
+            
+        # Parse hunk header
+        if line.startswith('@@'):
+            try:
+                # Extract new file line number (after +)
+                hunk_header = line.split('+', 1)[1].split(' ', 1)[0]
+                line_number = int(hunk_header.split(',')[0])
+                valid_lines[current_file].add(line_number)
+            except (IndexError, ValueError):
+                line_number = None  # Invalid hunk format
+            continue
+            
+        # Only process lines if we have valid line_number
+        if line_number is None:
+            continue
+            
+        # Track added lines
+        if line.startswith('+'):
+            valid_lines[current_file].add(line_number)
+            line_number += 1
+        # Track context lines (non-diff lines)
+        elif line.startswith(' '):
+            line_number += 1
+            
+    return valid_lines
+
+
+"""
+Posting can be done later when logs are correct and accurate. The line numbers are faulty and need to be fixed.
+"""
+
+
+def post_review_comment(comments, diff_content):
+    """Post validated review comments to GitHub"""
+    g = Github(os.getenv('GIT_TOKEN'))
+
+    # Load GitHub event data
+    event_path = os.getenv('GITHUB_EVENT_PATH')
+    with open(event_path, 'r') as f:
+        event_data = json.load(f)
+
+    # Get repository and PR details
+    repo = g.get_repo(event_data['repository']['full_name'])
+    pr_number = event_data['pull_request']['number']
+    pull_request = repo.get_pull(pr_number)
+    head_sha = event_data['pull_request']['head']['sha']
+
+    # Get valid lines from diff
+    valid_lines = get_valid_lines(diff_content)
+    
+    # Prepare filtered comments
+    filtered_comments = []
+    for comment in comments:
+
+        # Ensure we're dealing with a dictionary
+        if not isinstance(comment, dict):
+            print(f"Skipping invalid comment type: {type(comment)}")
+            continue
+
+        try:
+            file_path = comment['path']
+            line_num = comment.get('line', 1)
+            
+            # Validate against actual diff
+            if file_path in valid_lines and line_num in valid_lines[file_path]:
+                filtered_comments.append({
+                    "path": file_path,
+                    "position": line_num,  # GitHub expects 'position' not 'line'
+                    "body": comment['body']
+                })
```

Comment:
Consider using a more descriptive variable name than `e` for the KeyError exception in the comment validation.

```suggestion
except KeyError as key_err:
```